Upgrade helper version for hook in version 2.1

Josue-T opened the pull request:
## The problem

We still are using helper v2.0 for hooks

## Solution

Use helpers v2.1.

Note I specified the helper version in each file to avoid to upgrade the hook version in case some apps still use helper v2.0 for hooks.

## PR Status

Can be tested, seem working

## How to test

Run the regen-conf, backup, restore

Josue-T committed:
Upgrade helper version for hook in version 2.1

Josue-T committed:
Fix bash linter

Josue-T committed:
Fix shellcheck

Josue-T committed:
Fix backup/restore

Josue-T committed:
Fix shellcheck

Josue-T committed:
Fix tests

Josue-T committed:
Uniformize usage of set -Eeuo pipefail for all regen-conf

Josue-T committed:
Fix dnsmasq regen-conf

Josue-T committed:
Fix postinstall

zamentur commented on helpers/helpers.v2.1.d/backup:
```diff
@@ -88,7 +95,12 @@ ynh_backup() {
+    if [ "${YNH_CORE_SCRIPT:-}" != 1 ]; then
+        rel_dir="${YNH_CWD#"$YNH_BACKUP_DIR"}"
+    else
+        rel_dir="${YNH_CORE_BACKUP_DIR#"$YNH_BACKUP_DIR"}"
```
Use YNH_CWD instead of YNH_CORE_BACKUP_DIR like in https://doc.yunohost.org/admin/backups/include_exclude_files

Josue-T committed:
Remove YNH_CORE_SCRIPT env var

Salamandar commented:
This will break restoration of old backups.

Josue-T commented:
should be mostly good except one thing, how to handle the backup dir because currently we use YNH_CWD to define the destination and as the core backup is called all in one by as a hook we can't set YNH_CWD for each hook.

Josue-T commented:
> This will break restoration of old backups.

Well, normally no because the change try to stay retro compatible. And the original path is saved in the CSV, so at the end in most of case we don't really take care of where we backup. But fee free to bring some improvement in this PR if you have a better idea of how to handle this.

Josue-T committed:
Rework backup/restore hook to work without $YNH_CORE_BACKUP_DIR

Josue-T committed:
Use hook pre callback to define YNH_CWD for backup/restore hooks

Josue-T committed:
Merge branch 'dev' into hook_v21

Josue-T committed:
Fix config-regen postfix hook

Josue-T committed:
Fix helper call

Josue-T committed:
Fix unset var YNH_J2_FILTERS_FILE_PATH

Josue-T committed:
Fix missing variable set experimental in init conf regen

Josue-T committed:
Add missing all var for nginx config

Josue-T committed:
Fix ldap backup/restore

alexAubin committed:
hook.py: rename 'pre_callback' to 'compute_args_and_workdir_for_individual_script' such that it's actually meaningful ... + trash unused post_callback arg for hook_callback

alexAubin commented on helpers/helpers.v2.1.d/backup:
```diff
@@ -45,7 +45,14 @@ ynh_backup() {
+        # Apps backup
+        (-n "${app:-}" && "$target" =~ ^/var/log/$app/?\*?$) \
+        || (-n "${data_dir:-}" && "$target" =~ ^$data_dir) \
+        # for core backups
+        || (-z "${app:-}" && "$target" == '/home/'*) \
+        || (-z "${app:-}" && "$target" == '/var/mail') \
```
It's unclear why this was changed, this may change some existing behavior such as people calling something along the lines of:

```shell
BACKUP_CORE_ONLY=1 yunohost backup create
```

hoping to backup "everything" (apps and core) but without the app's core data ... with this change it would also exclude the /home/ and /var/mail dirs

(dunno how much of a big deal this is...)

alexAubin replied:
Update: supposedly that's because the core called `ynh_backup foo 1` with the "1" corresponding to the legacy "is_big" uuueargha

alexAubin commented on hooks/backup/18-data_multimedia:
```diff
@@ -22,15 +22,13 @@
 
 # Backup multimedia directory
-ynh_backup --src_path="/home/yunohost.multimedia" --dest_path="${backup_dir}" --is_big --not_mandatory
+ynh_backup "/home/yunohost.multimedia"
```
```suggestion
[ ! -e "/home/yunohost.multimedia" ] || ynh_backup "/home/yunohost.multimedia"
```

alexAubin committed:
Unecessary YNH_CWD

alexAubin committed:
Update 05-conf_ldap: unecessary YNH_CWD

alexAubin committed:
Update 05-conf_ldap: typo

alexAubin committed:
set -Eeuo pipefail is evil, cf https://mywiki.wooledge.org/BashPitfalls#pipefail

alexAubin committed:
helpers: simplify conditions when running helpers from the regenconf

alexAubin committed:
backup/restore hooks: misc cleanup/simplifications

alexAubin reviewed (approved):
(LGTM modulo how happy the tests are)

Josue-T committed:
Update hooks/backup/18-data_multimedia

Co-authored-by: Alexandre Aubin <4533074+alexAubin@users.noreply.github.com>

Josue-T committed:
Fix from linter

Josue-T committed:
Fix from linter

Josue-T committed:
Fix typo

Josue-T committed:
Merge dev into hook_v21

Josue-T committed:
Fix restore

Josue-T committed:
Fix restore

Josue-T committed:
Merge branch 'dev' into hook_v21

Josue-T commented:
@alexAubin do you think we could merge it maybe in Trixie ?

alexAubin reviewed (approved):
I'm gonna say it looks okay to me to merge in current `dev` but we may get some surprises even though we pretty much carefully reviewed it several times

Any opinion @zamentur @Salamandar ?

poireauu committed:
Translated using Weblate (Polish)

Currently translated at 62.9% (577 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

xabirequejo committed:
Translated using Weblate (Basque)

Currently translated at 100.0% (916 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/eu/

poireauu committed:
Translated using Weblate (Polish)

Currently translated at 67.2% (616 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

poireauu committed:
Translated using Weblate (Polish)

Currently translated at 68.8% (631 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

poireauu committed:
Translated using Weblate (Polish)

Currently translated at 69.9% (641 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

poireauu committed:
Translated using Weblate (Polish)

Currently translated at 71.2% (653 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

xmgz committed:
Translated using Weblate (Galician)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/gl/

Commit:
Translated using Weblate (Russian)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/ru/

Commit:
Translated using Weblate (Ukrainian)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/uk/

pp-r committed:
Translated using Weblate (French)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/fr/

poireauu committed:
Translated using Weblate (Polish)

Currently translated at 75.0% (688 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

poireauu committed:
Translated using Weblate (Polish)

Currently translated at 76.3% (700 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

gdhanush27 committed:
Added translation using Weblate (Tamil)

xabirequejo committed:
Translated using Weblate (Basque)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/eu/

Commit:
Translated using Weblate (Czech)

Currently translated at 19.8% (182 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

Commit:
Translated using Weblate (Czech)

Currently translated at 20.0% (184 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

Commit:
Translated using Weblate (Czech)

Currently translated at 20.1% (185 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

Commit:
Translated using Weblate (Czech)

Currently translated at 21.0% (193 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

Commit:
Translated using Weblate (Czech)

Currently translated at 23.1% (212 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

Commit:
Translated using Weblate (Czech)

Currently translated at 23.2% (213 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

Bruno-Van-den-Bosch committed:
Translated using Weblate (Dutch)

Currently translated at 21.1% (194 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/nl/

alexAubin committed:
security: handle the kernel special case when checking for security issues via the security.toml index (#2300)

alexAubin committed:
Translations update from Weblate (#2279)

* Translated using Weblate (Basque)

Currently translated at 100.0% (916 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/eu/

* Translated using Weblate (Polish)

Currently translated at 62.9% (577 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

* Translated using Weblate (Polish)

Currently translated at 67.2% (616 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

* Translated using Weblate (Polish)

Currently translated at 68.8% (631 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

* Translated using Weblate (Polish)

Currently translated at 69.9% (641 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

* Translated using Weblate (Polish)

Currently translated at 71.2% (653 of 916 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

* Translated using Weblate (Russian)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/ru/

* Translated using Weblate (Galician)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/gl/

* Translated using Weblate (Ukrainian)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/uk/

* Translated using Weblate (French)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/fr/

* Translated using Weblate (Polish)

Currently translated at 75.0% (688 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

* Translated using Weblate (Polish)

Currently translated at 76.3% (700 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/pl/

* Added translation using Weblate (Tamil)

* Translated using Weblate (Basque)

Currently translated at 100.0% (917 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/eu/

* Translated using Weblate (Czech)

Currently translated at 19.8% (182 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

* Translated using Weblate (Czech)

Currently translated at 20.0% (184 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

* Translated using Weblate (Czech)

Currently translated at 20.1% (185 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

* Translated using Weblate (Czech)

Currently translated at 21.0% (193 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

* Translated using Weblate (Czech)

Currently translated at 23.1% (212 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

* Translated using Weblate (Czech)

Currently translated at 23.2% (213 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/cs/

* Translated using Weblate (Dutch)

Currently translated at 21.1% (194 of 917 strings)

Translation: YunoHost/Core
Translate-URL: https://translate.yunohost.org/projects/yunohost/core/nl/

---------

Co-authored-by: xabirequejo <xabi.rn@gmail.com>
Co-authored-by: 1Poireau <unpoireau@proton.me>
Co-authored-by: Alekseï <support@digiology.org>
Co-authored-by: José M <correo@xmgz.eu>
Co-authored-by: ppr <pouetpouetratatouille@gmail.com>
Co-authored-by: Dhanush G <gdhanush270@gmail.com>
Co-authored-by: Jaroslav Nezbeda <jarda-oscloud@protonmail.com>
Co-authored-by: bruno van den bosch <brunovandenbosch@proton.me>

alexAubin committed:
Update changelog for 12.1.40

alexAubin committed:
diagnosis: fix fetching of the kernel version info, we want the mainline kernel version and not the debian kernel version

alexAubin committed:
diagnosis: fix confusing 'fixed in version' diagnosis message when it's in fact a dict

alexAubin committed:
Update changelog for 12.1.40.1

alexAubin committed:
:art: Format Python code with Ruff, and Bash code with Shfmt

Josue-T committed:
let's try to avoid having a shared dir for all config

Josue-T committed:
Merge branch 'dev' into hook_v21

Josue-T committed:
Try fix tests

Josue-T committed:
Try again

Josue-T committed:
We don't need set -eu because it's already handled by utils

Josue-T committed:
Fix tests

Josue-T committed:
Fix restore

zamentur commented:
About `set -eu` i found this commit https://github.com/YunoHost/yunohost/commit/1dece38a36bd58b3260902c6737498306bcdb219 . If i understand well `set` policy is now managed by 2.1 helpers, so may be we should discuss together to search if some things can break due to the new policy
